Structural code changes and CSV upload

souzamatheusp opened the pull request:
Uploaded IDE file by mistake. Going to delete it later.

souzamatheusp committed:
Added support for uploading CSV files
Added button and interface for uploading
Created new property (atos.csv) in each act. Null by default, uploaded file otherwise.
TODO:
Make it possible to upload more than one file
Make sure the file is *.csv (should any more files be allowed?). Can't do much about the content, I think
Tweak the interface
Reformat the code

souzamatheusp committed:
Structural code changes
Started to refactor the code
Started to introduce better object oriented code:
-Introduced UI object. It will, uh, hold the UI code. Doesn't include 'back front end', e.g. data managing.
Also fixed a minor bug where clicking in 'New Act' button when editing one didn't actually started a new one, rather proceeded to subscribe it

odanoburu commented:
boa, @souzamatheusp !

fiz um PR pro seu repo colocando os arquivos do `magnific-popup` e outras coisinhas. eu tô sem internet boa pra poder buildar a docker image, amanhã verei isso!

como a gente vai fazer pra receber o csv? mexemos no `main.py` pra ter um endereço de upload?

souzamatheusp committed:
Fixes
Committed missing dependencies
Removed deleted file
Removed IDE-specific file

souzamatheusp commented:
Arrumadas as dependências.
Sobre o csv: pensei um pouco, e achei dois métodos. Um, usar um request separado para mandar ele. Dois, embutir o csv no json.
Fui com o json pelo seguinte:
- Não requer nenhuma modificação no `main.py`. Nenhuma porta a mais, métodos etc.
- Possivelmente (provavelmente) haverá mais de um csv por DO. Isso iria requerer diversos requests ou um massivo. Em qualquer caso, teria de ser especificado para que ato o csv servirá. Então vem mais python para segurar os csv's, gerenciar e tudo mais. Mas já temos o json para essas coisas. Já fica organizado.
O que acha?

souzamatheusp committed:
Merge remote-tracking branch 'upstream/master'

souzamatheusp committed:
Further structural changes
-Continued object orienting. New object: DO. Will hold data managing code.
-Changed some variable names (to better ones)

odanoburu commented:
@souzamatheusp ótima ideia! facilita muito mesmo, não tem pq não fazer! pra vantagem é que fica mais fácil arquivar o DO (pra oferecer o download): só precisamos de um json por dia, nada de outros arquivos!

souzamatheusp committed:
Fix jQuery reference being made late
